### PR TITLE
feat(client): desktop version uses select2 to search, filter, and select

### PR DESCRIPTION
**Pull request by pAnhTri**
### What is the reason for a change?
I took a look at [issue #4346](https://github.com/SillyTavern/SillyTavern/issues/4346) which requests a filter/search functionality for lorebooks dropdown lists. The world info editor uses a native select element, which becomes tedious to navigate when users have many worlds loaded. This creates a poor user experience as users need to scroll through potentially hundreds of items to edit one specific lorebook/ world.

### What did you do to achieve this?
I referenced the select2 initiation of world info to replace the native select elements in the world info editor. The changes include:
- Initialize select2 for world info edit dropdown component
- Followed existing codebase with minimal changes for a non breaking implementation, maintaining all current functionalities and features

image

### How would a reviewer test the change?
1. **Setup**: Load or create multiple lorebooks/ worlds
2. **Test World Info Editor**:
   - Navigate to the world info editor
   - Verify the dropdown now shows a search input field
   - Type partial names of lorebooks and verify filtering works
   - Test keyboard navigation (arrow keys, enter, escape)
3. **Test Edge Cases**:
   - Try with very long lorebook names
   - Test with special characters in names
   - Verify behavior with empty search results
4. **Verify Existing Functionality**:
   - Ensure all existing binding functionality still works
   - Confirm no breaking changes to the world info system

## Type of Change
- [x] New feature (non-breaking change which adds functionality)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation update
- [ ] Performance improvement
- [ ] Code refactoring

## Testing
- [x] I have tested this change locally with multiple lorebooks
- [x] I have run `npm run lint` and fixed any errors
- [x] I have formatted the code using VS Code's autoformat
- [x] I have tested keyboard navigation and accessibility features

## Additional Notes
This implementation addresses the user experience issue described in issue #4346 by making it much easier to find and select specific lorebooks when users have many worlds loaded. The search functionality is case-insensitive and provides real-time filtering to improve workflow efficiency. There is now a search limit of 50 characters that can be removed if needed.

## Checklist:

- [x] I have read the [Contributing guidelines](https://github.com/SillyTavern/SillyTavern/blob/release/CONTRIBUTING.md).

### CODE REVIEW

File: public/scripts/world-info.js [hunk 1]
```diff
@@ -5718,6 +5718,49 @@ export function initWorldInfo() {
...
+    //**************************WORLD EDITOR SELECT*************************//
+    // Use select2 for the world editor select to filter and search for worlds to edit
+    // Makes it easier to find the world you want to edit in case of many worlds
+    // Limits the input length to 50 characters
+    //**************************WORLD EDITOR SELECT*************************//
+    if (!isMobile()) {
+        $('#world_editor_select').select2({
+            width: '100%',
+            placeholder: t`--- Pick to Edit ---`,
+            allowClear: false,
```
Comment (Cohee1207) [inline public/scripts/world-info.js]: How do you remove a selection when it's made?

File: public/scripts/world-info.js [hunk 2]
```diff
@@ -5718,6 +5718,49 @@ export function initWorldInfo() {
...
+            closeOnSelect: true,
+            multiple: false,
+            minimumInputLength: 0,
+            maximumInputLength: 50,
+            language: {
```
Comment (Cohee1207) [inline public/scripts/world-info.js]: I don't think you need any of those custom functions. The defaults are good enough

**Comment by Cohee1207** [comment]
The layout with full width selection looks awkward while the "New" button takes a full row. I'd swap them with the editor selection and place the "New" button on the editor buttons toolbar.

File: public/scripts/world-info.js (hunk 1 above)
Comment (pAnhTri) [reply public/scripts/world-info.js]: I initially thought that you wouldn't need to clear the selection since you can also choose a new selection to overwrite the previous one to edit.
My latest commit allows clearing of the selection.

File: public/scripts/world-info.js (hunk 2 above)
Comment (pAnhTri) [reply public/scripts/world-info.js]: Thank you for pointing that out. I removed the custom functions.

**Comment by pAnhTri** [comment]
> The layout with full width selection looks awkward while the "New" button takes a full row. I'd swap them with the editor selection and place the "New" button on the editor buttons toolbar.

I reduced the width to 30% to be more in line with what was used before:
image

**Comment by Cohee1207** [comment]
Updated the layout so that the toolbar row wouldn't feel so crammed, hence it's now split in two.
image
